src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3430,6 +3588,16 @@ internal void SetReceivingPacketInformation()
 
         internal unsafe void SetSocketOption(SocketOptionLevel optionLevel, SocketOptionName optionName, int optionValue, bool silent)
         {
+            // WASI is always set to receive PacketInformation
+            if (OperatingSystem.IsWasi() && optionName == SocketOptionName.PacketInformation)
+            {
+                if (optionValue == 0)
+                {
+                    UpdateStatusAfterSocketOptionErrorAndThrowException(SocketError.ProtocolOption);
```

Comment:
I'm not sure we need to throw if the platform does _more_ than asked for. If we have concerns we can can strip to info internally IMHO.

Reply:
done

Reply:
right. We can still simply throw it out. And not receiving the info is default AFAIK.  mostly perf improvement IMHo but that probably does not matter for your use case.